docs_src/design/adr/system-management/0006-Metrics-ControlPlaneEvents.md

```diff
@@ -0,0 +1,226 @@
+
+In the future, EdgeX may consume its own metric data or CPE.  For example, EdgeX may, in the future,
+
+- use a metric on the number of EdgeX events being sent to core data as the means to throttle back device data collection
+- use a configuration change CPE to automatically restart a service.
+
+In the future, EdgeX application services may optionally subscribe to metrics and CPE messages (by attaching to the appropriate message pipe).  Thus allowing additional filtering, transformation, endpoint control of metric and CPE data.
+
+- At the point where this feature is supported, consideration would need to be made as to whether all events (sensor reading messages and metric/CPE messages go through the same application services)
+
+At this time, EdgeX will not persist the metric or CPE data (except as it may be retained as part of message delivery).  Consumers of metric or CPE data are responsible for persisting the data if needed.  Persistence of metric or CPE information may be considered in the future.
+
+In general, EdgeX metrics and CPE are meant to provide external applications and systems better information about what is happening "inside" EdgeX and the devices with which it communicates.
+
+### Requirements
+
+- Services will push specified metrics collected for that service to a specified (by configuration) message endpoint (MQTT topic, 0MQ topic or other EdgeX supported messaging implementation)
+  - Use and configuration to target a secure endpoint (MQTTS) must be provided.  As exemplified by application services, there must be the option to use "insecure secrets" or Vault-backed security services when a secure endpoint is desired.
+- Services will push all CPE detected for that service to a specified (by configuration) message endpoint.
+  - The endpoint for metric and CPE may be the same or different at the user's discretion.
+  - Use and configuration to target a secure endpoint (MQTTS) must be provided.  As exemplified by application services, there must be the option to use "insecure secrets" or Vault-backed security services when a secure endpoint is desired.
+- All services must document what metrics and CPE they offer.
+- Services must provide REST endpoints that respond with what metrics and CPE the service offers (and whether that metric or CPE is currently `on` or `off`).
+  - Proposed endpoint format:  /api/v2/metrics and /api/v2/cpe
+- Services will have configuration which allows EdgeX system managers to select which metrics and CPE are turned `on` or `off`.
+  - When a metric or CPE is turned `off` the service does not collect the metric or monitor for the CPE.  When a metric is turned `on` the service collects and sends the metric to the designated endpoint.  When a CPE is turned `on` the service monitors for the event and sends any detected CPE to the designated endpoint.

```

Comment:
Here you are using `designated endpoint` rather than message bus. Need to be consistently be `sends to message bus`

Reply:
good comments.  I had not thought of categorizing metrics/CPE but I suppose there may be logical groupings that this would add user convenience (but add to implementation complexity).  

Again, the endpoints are just for getting status of metrics/CPE collection/monitoring and not where the data is sent.